File: js/stack.js
class Node{
	constructor(val,next){
		this.val = val;
		this.next = next;
	}
}

class stk{
	constructor(){
		let top = null;
		let bottom = null;
		let size = 0;
	}

	push(val){
		let nn = new Node(val,null);
		if(this.top != null)
			nn.next = this.top;
		else
			this.bottom = nn;
		this.top = nn;
		this.size++;
	}

	display(){
		let p = this.top;
		while(p!=null){
			console.log(p.val);
			p = p.next;
		}
	}

	pop(){
		let rv = this.top;
		this.top = this.top.next;
		this.size--;
		return rv.val; 
	}

	isempy(){
		return this.head == null;
	}
}

let s = new stk();
s.push(10);
s.push(20);
s.push(30);
s.display();
console.log('-------------------\n');
console.log(s.pop());
console.log('-------------------\n');
s.display();

File: cpp/stack.cpp
#include <bits/stdc++.h>
using namespace std;

struct Node{
	int val;
	Node* next;
};
class stk{
	public:
		Node*top = NULL;
		Node*bottom = NULL;
		int size = 0;

		void push(int val){
			Node* nn = new Node();
			nn->val = val;
			if(top != NULL)
				nn->next = top;
			else 
				bottom = nn;
			top = nn;
			size++;
		}

		void display(){
			Node*p = top;
			while(p!=NULL){
				cout<<p->val<<endl;
				p = p->next;
			}
		}

		int pop(){
			Node* rv = top;
			top = top->next;
			rv->next = NULL;
			size--;
			return rv->val;
		}

		int isempy(){
			return top == NULL;
		}

};
int main(int argc, char const *argv[])
{
	stk s ;
	s.push(10);
	s.push(20);
	s.push(30);
	s.display();
	cout<<"-------"<<endl;
	cout<<s.pop()<<endl;
	cout<<"-------"<<endl;
	s.display();
	return 0;
}